src/contracts/services/lend.ts

```diff
@@ -0,0 +1,512 @@
+import {
+  BatchQueryParams,
+  Contract,
+  BatchQueryParsedResponse,
+  MinBlockHeightValidationOptions,
+} from '~/types';
+import { batchQuery$ } from '~/contracts/services/batchQuery';
+import {
+  map,
+  first,
+  switchMap,
+  lastValueFrom,
+} from 'rxjs';
+import {
+  VaultResponse,
+  VaultsResponse,
+  NormalizationFactor,
+  PositionResponse,
+  PositionsResponse,
+} from '~/types/contracts/lend/response';
+import {
+  convertCoinFromUDenom,
+  getActiveQueryClient$,
+} from '~/index';
+import {
+  Vaults,
+  Vault,
+  BatchVaults,
+  VaultUserData,
+  VaultsUserData,
+  BatchVaultsUserData,
+  VaultVersion,
+  LendVaultRegistryContract,
+} from '~/types/contracts/lend/model';
+import {
+  msgGetVault,
+  msgGetVaults,
+  msgGetVaultUserPosition,
+  msgGetVaultUserPositions,
+} from '~/contracts/definitions/lend';
+import { sendSecretClientContractQuery$ } from '~/client/services/clientServices';
+import { AccountPermit } from '~/types/permit';
+import BigNumber from 'bignumber.js';
+
+/**
+* Parse lend vault response
+*/
+function parseLendVault(vault: VaultResponse, vaultVersion: VaultVersion) {
+  const {
+    id,
+    allowance,
+    collateral: {
+      elastic: elasticCollateral,
+      base: baseCollateral,
+      last_accrued: collateralLastAccrued,
+    },
+    safe_collateral: safeCollateral,
+    config: {
+      max_ltv: maxLtv,
+      collateral_oracle_delay: collateralOracleDelay,
+      fees: {
+        interest_rate: interestRate,
+        borrow_fee: borrowFee,
+        liquidation_fee: liquidationFee,
+      },
+    },
+    debt,
+    collateral_addr: collateralAddress,
+    is_protocol: isProtocol,
+    name,
+    open_positions: openPositions,
+    position_id_counter: positionIdCounter,
+  } = vault.vault;
+
+  return {
+    id,
+    vaultVersion,
+    name,
+    collateralAddress,
+    silkMaxAllowance: convertCoinFromUDenom(allowance.max, NormalizationFactor.LEND).toString(),
+    silkAllowanceUsed: convertCoinFromUDenom(allowance.used, NormalizationFactor.LEND).toString(),
+    maxLtv: Number(maxLtv),
+    // Collateral is expressed differently depending on vault type
+    collateral: {
+      total:
+    (vaultVersion === VaultVersion.V1 || vaultVersion === VaultVersion.V2)
```

Comment:
Is there a reason that V2 is included in this check? Why did V3 get the big change?

Reply:
it's different math for the collateral display. I don't know why it was such a big change, i wish it would have been consistent.